jlc_kicad_tools/generate_jlc_files.py

```diff
@@ -20,16 +20,38 @@
 import re
 import sys
 import argparse
-import logging
 import errno
 
+from jlc_kicad_tools.logger import Log
 from jlc_kicad_tools.jlc_lib.cpl_fix_rotations import ReadDB, FixRotations
 from jlc_kicad_tools.jlc_lib.generate_bom import GenerateBOM
 
 DEFAULT_DB_PATH="cpl_rotations_db.csv"
 
-def main():
-	parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter, description='Generates BOM and CPL in CSV fashion to be used in JLCPCB Assembly Service', prog='generate_jlc_files')
+_LOGGER = Log()
+
+class Component():
```

Comment:
Your thoughts are completely valid. I decided just to take off `Component` class it all since after reading the pros and cons is just make me realize that it was not a good strategy in the long run. 

About `isPathFound()`, let's leave logic as it was since I don't want to introduce by any means code that it's not efficient even though we have the certainty that this is not a performance-critical use case.